apm-agent-plugins/apm-log-shader-plugin/apm-logback-plugin/apm-logback-plugin-impl/src/main/java/co/elastic/apm/agent/logback/LogbackAppenderAppendAdvice.java

```diff
@@ -36,19 +35,19 @@
     @Advice.OnMethodEnter(suppress = Throwable.class, skipOn = Advice.OnNonDefaultValue.class, inline = false)
     public static boolean shadeAndSkipIfOverrideEnabled(@Advice.Argument(value = 0, typing = Assigner.Typing.DYNAMIC) final Object eventObject,
                                                         @Advice.This(typing = Assigner.Typing.DYNAMIC) OutputStreamAppender<ILoggingEvent> thisAppender) {
-        return thisAppender instanceof FileAppender && eventObject instanceof ILoggingEvent &&
-            LogbackLogShadingHelper.instance().shouldSkipAppend((FileAppender<ILoggingEvent>) thisAppender);
+
+        return eventObject instanceof ILoggingEvent && LogbackEcsReformattingHelper.instance().onAppendEnter(thisAppender);
```

Comment:
It's a pitty that we can't use `@Local` in indy advices. Propagating the config value from the enter to the exit advice that way would be neat. But ofc. the ThreadLocal also works fine.

Reply:
It wouldn't be enough in this case though, because in log4j2 we need to use it in a separate advice as well